server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -145,8 +145,10 @@ public class InternalEngine extends Engine {
     private final LiveVersionMapArchive liveVersionMapArchive;
     // Records the last known generation during which LiveVersionMap was in unsafe mode. This indicates that only after this
     // generation it is safe to rely on the LiveVersionMap for a real-time get.
-    // TODO: can we move this entirely into the stateless plugin?
+    // TODO: move the following two to the stateless plugin
     private final AtomicLong lastUnsafeSegmentGenerationForGets = new AtomicLong(-1);
+    // Records the segment generation for the currently ongoing commit if any, or the last finished commit otherwise.
+    protected final AtomicLong preCommitRecordedSegmentGeneration = new AtomicLong(-1);
```

Comment:
Can we make this private instead and add a protected accessor for the long value instead?